storage_kvm.py

```diff
@@ -225,21 +254,51 @@ def read(
         3021: f'3021: Failed to connect to the host {host}',
         3022: f'3022: Failed to read storage_kvm {domain_path}{storage} on the host {host}'
     }
-
+    message_list = {}
+    data_dict = {
+        host: {
+            'image': None,
+            'disk_size': None,
+        }
+    }
     # Define payload
     payload = f'qemu-img info {domain_path}{storage}'
 
     # Read storage using SSH communication
-    try:
-        exit_code, stdout, stderr = comms_ssh(
-            host_ip=host,
-            payload=payload,
-            username='robot',
+    response = comms_ssh(
+                host_ip=host,
+                payload=payload,
+                username='robot',
         )
-    except CouldNotConnectException:
-        return False, messages[3021]
-
-    if exit_code != SUCCESS_CODE:
-        return False, f'{messages[3022]} {exit_code}\nSTDOUT: {stdout}\nSTDERR: {stderr}'
-
-    return True, f'{messages[1000]}.\n STDOUT from the Host {host} is {stdout}'
+    if response['channel_code'] != CHANNEL_SUCCESS:
+        msg = f'{messages[3021]}\n ' \
+              f'channel_code: {response["channel_code"]}s.\n' \
+              f'channel_message: {response["channel_message"]}\n' \
+              f'channel_error: {response["channel_error"]}'
+        message_list.append(msg)
+        success = False
+    if response['payload_code'] != SUCCESS_CODE:
+        msg = f'{messages[3022]}\n ' \
+              f'payload_code: {response["payload_code"]}s.\n' \
+              f'payload_message: {response["payload_message"]}\n' \
+              f'payload_error: {response["payload_error"]}'
+        message_list.append(msg)
+        success = False
+
+    if success:
+        stdout = response.get('stdout', '')
+
+        # Extract the image path
+        image_match = re.search(r'image: (\S+)', stdout)
+        image = image_match.group(1) if image_match else None
+
+        # Extract the disk size
+        disk_size_match = re.search(r'disk size: (\S+)', stdout)
```

Comment:
The output of the payload looks like:
```
image: /var/lib/libvirt/images/test123.img
file format: qcow2
virtual size: 20 GiB (21474836480 bytes)
disk size: 196 KiB
cluster_size: 65536
Format specific information:
    compat: 1.1
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
```

We need to the `virtual size` not the disk size, so can you correct this to collect the virtual size as just `size` as commented above.

And also collect the only value with GiB ie as per above output the data_dict would be :
`
data_dict = {
     'image': '/var/lib/libvirt/images/test123.img',
     'size': 20,
 }
`